crates/bevy_ecs/src/resource.rs

```diff
@@ -72,4 +86,123 @@ pub use bevy_ecs_macros::Resource;
     label = "invalid `Resource`",
     note = "consider annotating `{Self}` with `#[derive(Resource)]`"
 )]
-pub trait Resource: Send + Sync + 'static {}
+pub trait Resource: Component<Mutability = Mutable> {
```

Comment:
To confirm the design tradeoff we want here: Making `trait Resource: Component` means it's valid to 

```rust
#[derive(Resource)]
struct R;

world.spawn(R);
```

Due to the hooks, that will work and create a new resource the first time it's run, and log a warning and spawn an empty entity if it's run multiple times.

Reply:
Yes